packages/webdriverio/src/scripts/resq.js

```diff
@@ -0,0 +1,55 @@
+export const waitToLoadReact = function waitToLoadReact () {
+    window.resq.waitToLoadReact()
+}
+
+export const react$ = function react$ (selector, props, state) {
+    let elems = window.resq.resq$$(selector)
+
+    /**
+     * see https://github.com/baruchvlz/resq/issues/19
+     */
+    if (typeof elems === 'string') {
+        throw new Error(elems)
+    }
+
+    elems = elems
+        .byProps(props)
+        .byState(state)
```

Comment:
Hm, I don't think this will work as expected.

Take this scenario

```javascript
return (
  <div>
    <MyCmp s="b" />
    <MyCmp s="b" />
  </div>
);
```

Searching `byProps({})` we would expect to return the component, but currently this is the behaviour:

```javascript
resq$$('MyCmp').byProps({}).length // 0
resq$$('MyCmp').byProps({ s: 'b' }).length // 2
```

I think I'll release a patch later today to fix this so that if we pass an empty object the filter should work as if it's not being filtered at all. What do you think?

Reply:
> I think I'll release a patch later today to fix this so that if we pass an empty object the filter should work as if it's not being filtered at all. What do you think?

Yeah I think that would make sense. If the behavior is that if you pass in a filter like `{a: 'foobar'}` it finds components that have a property `a` equals `foobar` __and__ also others rather than finding only components that contain only on property `a` equals `foobar`. strict vs non strict and I think if you pass in a filter atm it is non strict so should it behave passing an empty object.